### PR TITLE
Project refactoring

### PR DESCRIPTION
**Pull request by rec**
These changes are on the way to the correct "default driver/layout/etc" idea.

There are two things going on here!

First is that my whole idea of "settings" is wrong.  So I killed it.  :-D  We'll be using just `DataFile` and life will be much simpler.

Then before I went ahead, I cleaned up project.py [considerably](https://github.com/ManiacalLabs/BiblioPixel/compare/dev...rec:project-refactoring#diff-398a55c2da42aa3c07fa0ba2a0f1edde) and fixed a few other minor issues.

### CODE REVIEW

File: bibliopixel/animation/animation.py [hunk 1]
```diff
@@ -138,15 +138,15 @@ def run_all_frames(self):
                 self.run_one_frame()
 
     def set_runner(self, runner):
-        self.runner = runner
+        self.runner = Runner(**(runner or {}))
```
Comment (adammhaile) [inline bibliopixel/animation/animation.py]: Why? Weren't we passing in a complete Runner() before?

File: bibliopixel/animation/animation.py [hunk 2]
```diff
@@ -138,15 +138,15 @@ def run_all_frames(self):
...
         self.threading = AnimationThreading(self.runner, self.run_all_frames)
 
     def start(self):
         self.threading.start()
 
     def run(self, **kwds):
         # DEPRECATED
-        self.set_runner(Runner(**kwds))
+        self.set_runner(kwds)
```
Comment (adammhaile) [inline bibliopixel/animation/animation.py]: Oh... I see. Still, why the change?

File: bibliopixel/layout/matrix.py [hunk 3]
```diff
@@ -16,8 +16,8 @@ def __init__(self, drivers, width=0, height=0, coordMap=None,
         driver - instance that inherits from DriverBase
         width - X axis size of matrix
         height - Y axis size of matrix
-        coordMap - a 2D matrix defining the X,Y to strip index mapping. Not needed in most cases
-        rotation - how to rotate when generating the map. Not used if coordMap specified
+        coord_map - a 2D matrix defining the X,Y to strip index mapping. Not needed in most cases
+        rotation - how to rotate when generating the map. Not used if coord_map specified
```
Comment (adammhaile) [inline bibliopixel/layout/matrix.py]: You're breaking 3.0.5 compat... I mean, yeah, no one probably uses it directly, just wanted to note it.

File: bibliopixel/layout/multimap.py [hunk 4]
```diff
@@ -1,10 +1,33 @@
...
+    def _make_driver(self, width, height, matrix=None, **kwds):
+        row = gen_matrix(width, height, **(matrix or {}))
+        for y, row_entry in enumerate(row):
+            self.map.append([i + self.offset for i in row_entry])
+
+        self.offset += len(row) * len(row[0])
+        return self.make_object(width=width, height=height, **kwds)
 
     def addRow(self, *maps):
+        # DEPRECATED
```
Comment (adammhaile) [inline bibliopixel/layout/multimap.py]: Why?

File: bibliopixel/project/project.py [hunk 5]
```diff
@@ -1,67 +1,54 @@
...
+        raise ValueError('layout was not specified in project')
+
+    if not (driver or drivers):
+        raise ValueError('Projects has neither driver nor drivers sections')
+
+    gitty.sys_path.extend(path or '')
+    maker = data_maker.Maker(**(maker or {}))
+    make_object = functools.partial(importer.make_object, maker=maker)
+
+    builder = MultiMapBuilder(make_object)
```
Comment (adammhaile) [inline bibliopixel/project/project.py]: We still need to make all this easier in project files.

File: bibliopixel/animation/animation.py (hunk 2 above)
Comment (rec) [inline bibliopixel/animation/animation.py]: Why should other classes need to know about the existence of Runner, which they never see?

File: bibliopixel/layout/matrix.py (hunk 3 above)
Comment (rec) [inline bibliopixel/layout/matrix.py]: Yes, that's true.  :-/

File: bibliopixel/layout/multimap.py (hunk 4 above)
Comment (rec) [inline bibliopixel/layout/multimap.py]: `add_one_row` completely replaces `addRow`.

In practice `len(maps)` is always 1. Nearly all that code is involved with keeping track of the cases where `len(maps) > 1`.

(More, calling `add_one_row` repeatedly has exactly the same effect that `addRow` would have had with `maps` of length greater than 1.  !)

File: bibliopixel/project/project.py (hunk 5 above)
Comment (rec) [inline bibliopixel/project/project.py]: Agree 100% - it's also necessary for #270.

File: bibliopixel/animation/animation.py (hunk 2 above)
Comment (adammhaile) [inline bibliopixel/animation/animation.py]: Fair enough.

File: bibliopixel/layout/multimap.py (hunk 4 above)
Comment (adammhaile) [inline bibliopixel/layout/multimap.py]: +1
